File: cryze_android_app/android/app/src/main/java/com/tencentcs/iotvideo/iotvideoplayer/ConnectMode.java
package com.tencentcs.iotvideo.iotvideoplayer;

public class ConnectMode {
    public int mMode = 0;
    public int mProtocol = -1;

    public interface Mode {
        public static final int LAN = 3;
        public static final int P2P = 2;
        public static final int RELAY = 1;
        public static final int UNINITIALIZED = -1;
        public static final int UNKNOWN = 0;
    }

    public interface Protocol {
        public static final int TCP = 1;
        public static final int UDP = 0;
        public static final int UNKNOWN = -1;
    }

    private String getModeString(int mode) {
        switch (mode) {
            case Mode.LAN:
                return "LAN";
            case Mode.P2P:
                return "P2P";
            case Mode.RELAY:
                return "RELAY";
            case Mode.UNINITIALIZED:
                return "UNINITIALIZED";
            case Mode.UNKNOWN:
            default:
                return "UNKNOWN";
        }
    }

    private String getProtocolString(int protocol) {
        switch (protocol) {
            case Protocol.TCP:
                return "TCP";
            case Protocol.UDP:
                return "UDP";
            case Protocol.UNKNOWN:
            default:
                return "UNKNOWN";
        }
    }

    public String toString() {
        StringBuilder sb2 = new StringBuilder("ConnectMode{mMode=");
        sb2.append(getModeString(this.mMode));
        sb2.append(", mProtocol=");
        sb2.append(getProtocolString(this.mProtocol));
        sb2.append('}');
        return sb2.toString();
    }
}

File: cryze_android_app/android/app/src/main/java/com/tencentcs/iotvideo/iotvideoplayer/ConnectMode.kt
package com.tencentcs.iotvideo.iotvideoplayer

enum class Mode(val value: Int) {
    LAN(3),
    P2P(2),
    RELAY(1),
    UNINITIALIZED(-1),
    UNKNOWN(0);

    companion object {
        fun fromValue(value: Int): Mode {
            return entries.find { it.value == value } ?: UNKNOWN
        }
    }
}

enum class Protocol(val value: Int) {
    TCP(1),
    UDP(0),
    UNKNOWN(-1);

    companion object {
        fun fromValue(value: Int): Protocol {
            return entries.find { it.value == value } ?: UNKNOWN
        }
    }
}

class ConnectMode(connectionMode: IntArray?) {
    // Leave this open (not private) for future use
    var mMode = Mode.UNKNOWN
        private set

    // Leave this open (not private) for future use
    var mProtocol = Protocol.UNKNOWN
        private set

    init {
        if (connectionMode != null && connectionMode.size == 3) {
            val i = connectionMode[0]
            if (i == 0) {
                this.mMode = Mode.fromValue(connectionMode[1])
                this.mProtocol = Protocol.fromValue(connectionMode[2])
            } else {
                this.mMode = Mode.fromValue(i)
            }
        }
    }

    override fun toString(): String {
        val sb2 = StringBuilder("ConnectMode{mMode=")
        sb2.append(this.mMode)
        sb2.append(", mProtocol=")
        sb2.append(this.mProtocol)
        sb2.append('}')
        return sb2.toString()
    }
}
